node/src/cli.rs

```diff
@@ -56,47 +54,6 @@ pub enum Subcommand {
 	/// The pallet benchmarking moved to the `pallet` sub-command.
 	#[command(subcommand)]
 	Benchmark(frame_benchmarking_cli::BenchmarkCmd),
-
-	/// Try some experimental command on the runtime. This includes migration
-	/// and runtime-upgrade testing.
-	#[cfg(feature = "try-runtime")]
-	TryRuntime(try_runtime_cli::TryRuntimeCmd),
-}
-
-/// Command for exporting the genesis state of the parachain
-#[derive(Debug, Parser)]
-pub struct ExportGenesisStateCommand {
-	/// Output file name or stdout if unspecified.
-	#[clap(value_parser)]
-	pub output: Option<PathBuf>,
-
-	/// Write output in binary. Default is to write in hex.
-	#[clap(short, long)]
-	pub raw: bool,
-
-	/// Id of the parachain this state is for.
-	#[clap(long)]
-	pub parachain_id: Option<u32>,
-
-	/// The name of the chain for that the genesis state should be exported.
-	#[clap(long)]
-	pub chain: Option<String>,
-}
-
-/// Command for exporting the genesis wasm file.
-#[derive(Debug, Parser)]
-pub struct ExportGenesisWasmCommand {
-	/// Output file name or stdout if unspecified.
-	#[clap(value_parser)]
-	pub output: Option<PathBuf>,
-
-	/// Write output in binary. Default is to write in hex.
-	#[clap(short, long)]
-	pub raw: bool,
-
-	/// The name of the chain for that the genesis wasm file should be exported.
-	#[clap(long)]
-	pub chain: Option<String>,
 }
```

Comment:
But the CLI only is for try runtime, right? This here is needed in order to generate chain-specs and getting genesis states.

Reply:
I am investigating these currently and we definitely need to fix something because the exported genesis state is not deterministic, i.e. 2 exports don't yield the same genesis hex. Once I have a fix, I will open a PR against this branch. This will include other potential client fixes from local testing.

Reply:
For `try-runtime` we use a binary. But for these, we do not need to copy them because those structs have been offered as a lib, we can import them. See this: https://github.com/centrifuge/centrifuge-chain/pull/1756/files#diff-76397c90c6ee2b6b98efd989d3ae3c11e0bde3db30b5ef46f2762180f56041a8R27

Reply:
Right now, running a local chain does not work.. We definitely need a few changes before we merge this regarding the dropped CLI support for the `parachain-id` option when exporting the genesis state or WASM. It is expected to be part of the chain spec and I don't think we should add support for that because the Substrate CLI for spec building has not supported providing the parachain ID via the CLI for quite some time.

We only need this feature for ephemeral parachains. My proposal is to hardcode a parachain id for those cases which would be low effort solution. I don't think we need to spawn a locally build parachain node that often. Only for rare testing cases like these. This used to be handy for testing runtime upgrades. Now Chopsticks is much more convenient in 99% of the cases because RUs on empty parachains are less meaningful.

Reply:
I agree with simplifying things. Thanks for diving into this ❤️